tempesta_fw/classifier/frang.c

```diff
@@ -262,6 +295,12 @@ frang_conn_close(struct sock *sk)
 }
 
 static int
+frang_time_in_frame(const unsigned long tcur, const unsigned long tprev)
+{
+	return tprev + FRANG_FREQ > tcur;
```

Comment:
The function works wrong in `frang_bad_resp_limit()` and right in `frang_req_limit()`.  Currently the function checks whether the difference between `tpev` and `tcur` is less than one second. The function must be extended with third argument: time frame in seconds, which must be used as multiplier  for `FRANG_FREQ`.

Reply:
```
ts = frang_resp_quantum();
i = ts % FRANG_FREQ;
if (ts != stat[i].ts) {
    stat[i].ts = ts;
    stat[i].cnt = 0;
}
ts = frang_resp_quantum() == ((jiffies / HZ) % FRANG_RESP_TIME_PERIOD) * FRANG_FREQ / frang_cfg.http_resp_code_block->tf;
```
In other words:
```
ts = (seconds % period) / (time_frame / FRANG_FREQ) == int32_seconds / (time_frame / FRANG_FREQ)
```
Let tf = 80, FRANG_FREQ = 8, and (jiffies / HZ) % FRANG_RESP_TIME_PERIOD == 100
0th second: ts0 = floor(100 / (80/8)) = 10
10th second: ts10 = floor(110 / (80/8)) = 11
80th second: ts80 = floor(180 / (80/8)) = 18

ts80 - ts0 == 8 >= FRANG_FREQ
80th second is in the other time frame than 0th second.

Reply:
Yes, i see, it was mine mistake. Sorry for that. The patch set can be merged into master.